lib/engine/game/base.rb

```diff
@@ -654,7 +663,16 @@ def operated_operators
       end
 
       def current_action_id
-        @actions.size + 1
+        @actions.size
```

Comment:
Because of the hashes in the array, that didn't work. :-/. I tried a few ways to work around that and they all make this function  ugly because the result can be an action, hash, or nil. I can change if you'd prefer `&.id` and are okay with a more complex function.

Reply:
why don't you fix the hash issue

Reply:
Fixed.